calculator/main.py

```diff
@@ -26,8 +33,17 @@ def perform_operation(calc, operation, params):
     :param params: Tuple containing the list of nums to operate with.
                    ie: (1, 2, 3, 4.5, -2)
     """
-    pass
-
+    for i in params:
+        if not isinstance(i, (int,float)):
+            raise InvalidParams("Given params are invalid.")
+    
+    total = calc['operations'][operation](*params)
+
+    calc['history'].append((str(datetime.now()),operation,params,total))
```

Comment:
Separate the arguments with an space. It's just a formatting/style from PEP8: https://www.python.org/dev/peps/pep-0008/#whitespace-in-expressions-and-statements

```python
calc['history'].append((str(datetime.now()), operation, params, total))
```